shiny/Data-Ingest/helper.R

```diff
@@ -0,0 +1,29 @@
+# This helper file stores all the relevant functions to be called in the shiny app
+#' Auto-create directory name for local file upload
+#'
+#' @param format_name Format name in type: character string
+#' @param site_id Integer, scientific notation, or character string that will be converted to an integer and will be the trailing identifier for the firectory name
+#'
+#' @return Self-generated directory name to store files uploaded via local upload. 
+#' @export
+#' @author Liam Burke (liam.burke24@gmail.com)
+#'
+#' @examples
+#' auto.name.directory(format_name = "LTER-hf-103", site_id = 1000004955)
+auto.name.directory <- function(format_name, site_id){
+  
+  # Split the format_name by the first non-alphanumeric character
+  splits <- base::strsplit("LTER-hf-103", "[^a-zA-Z0-9 :]")
+  
+  # Grab first split
+  basename <- splits[[1]][1]
```

Comment:
I don't see the need for this -- parsing out parts of a format name to shorten it looks like you're putting active effort into creating ambiguity. Just use the whole format name (though it would be good to make sure whitespace and other possible troublesome characters, e.g. `/`, are removed)

Reply:
Ok. The thought was to try an prevent long/ complicated names. I can change this.